### PR TITLE
Hints: add hint prompt attempts threshold override field for levelbuilders

### PR DESCRIPTION
**Pull request by Erin007**
[STAR-1205](https://codedotorg.atlassian.net/browse/STAR-1205)
Follow up to #36488 and #36507 

We suspect that the default `hint_prompt_attempts_threshold` will be good enough (at least for the interim) for most levels, but we might find the 7 attempts threshold for some levels to be wildly too high or too low.  In those cases, levelbuilders can re-set the `hint_prompt_attempts_threshold` to a more appropriate number. 

Screen Shot 2020-09-03 at 2 39 41 PM

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/models/levels/level.rb [hunk 1]
```diff
@@ -730,6 +730,12 @@ def all_child_levels
     (contained_levels + [project_template_level] - [self]).compact
   end
 
+  # There's a bit of trickery here. We consider a level to be
+  # hint_prompt_enabled for the sake of the level editing experience if any of # the scripts associated with the level are hint_prompt_enabled.
```
Comment (Hamms) [inline dashboard/app/models/levels/level.rb]:
```suggestion
  # hint_prompt_enabled for the sake of the level editing experience if any of
  # the scripts associated with the level are hint_prompt_enabled.
```

File: dashboard/app/models/levels/level.rb [hunk 2]
```diff
@@ -730,6 +730,12 @@ def all_child_levels
...
+  def hint_prompt_enabled?
+    !script_levels.map(&:script).select(&:hint_prompt_enabled?).empty?
```
Comment (Hamms) [inline dashboard/app/models/levels/level.rb]:
```suggestion
    script_levels.map(&:script).select(&:hint_prompt_enabled?).any?
```

**Review by Hamms — APPROVED** [review]
I would note that gating the form field behind the level being hint prompt enabled does mean that the level will need to be created, then added to a script, then edited by a levelbuilder in that order. It'd probably be worth checking in with levelbuilders to see how onerous that requirement is (particularly when compared to the potential for confusion if we show the form field for all levels, even those for which it's not applicable)

Aside from that usability concern and a couple nits, this LGTM!

**Review by dmcavoy — APPROVED** [review]
LGTM

**Comment by Erin007** [comment]
> I would note that gating the form field behind the level being hint prompt enabled does mean that the level will need to be created, then added to a script, then edited by a levelbuilder in that order. It'd probably be worth checking in with levelbuilders to see how onerous that requirement is (particularly when compared to the potential for confusion if we show the form field for all levels, even those for which it's not applicable)

Levelbuilders are 👍  for this change based on the assumption that we'll only want to adjust the `hint_promp_attempts_threshold` for levels that have been out in the wild long enough for us to determine that the default is far off.  It's not a field they expect to care much about when a level is first being built or not yet used in a script.

**Pull request merged**